### PR TITLE
gh-124412: Add helpers for converting annotations to source format

### PR DESCRIPTION
**Pull request by JelleZijlstra**
This is on top of #124479 because that PR changes some related logic.

* Issue: gh-124412

----
📚 Documentation preview 📚: https://cpython-previews--124551.org.readthedocs.build/

### CODE REVIEW

File: Lib/annotationlib.py [hunk 1]
```diff
@@ -750,3 +762,57 @@ def get_annotations(
...
+        return f"{value.__module__}.{value.__qualname__}"
+    if value is ...:
+        return "..."
+    if isinstance(value, (types.FunctionType, types.BuiltinFunctionType)):
+        return value.__name__
+    return repr(value)
+
+
+def annotations_to_source(annotations):
+    """Convert an annotation dict containing values to approximately the SOURCE format."""
```
Comment (carljm) [inline Lib/annotationlib.py:787]: "approximately the SOURCE format" 😆

**Comment by carljm** [comment]
Can you rebase this so the changes from #124479 are no longer included in it?

File: Lib/typing.py [hunk 2]
```diff
@@ -242,21 +242,10 @@ def _type_repr(obj):
...
-        if obj.__module__ == 'builtins':
-            return obj.__qualname__
-        return f'{obj.__module__}.{obj.__qualname__}'
-    if obj is ...:
-        return '...'
-    if isinstance(obj, types.FunctionType):
-        return obj.__name__
     if isinstance(obj, tuple):
         # Special case for `repr` of types with `ParamSpec`:
         return '[' + ', '.join(_type_repr(t) for t in obj) + ']'
```
Comment (carljm) [inline Lib/typing.py:247]: This is a case we don't want in `value_to_source`?

**Review by carljm — APPROVED** [review]

File: Lib/typing.py (hunk 2 above)
Comment (JelleZijlstra) [reply Lib/typing.py:247]: Yes, I think it's specifically for the internals of Callable and doesn't make sense in general. (Notice that it represents tuples with list syntax.)

**Pull request merged**